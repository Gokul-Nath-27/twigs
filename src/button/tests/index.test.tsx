import React from 'react';
import { createRoot } from 'react-dom/client';
import 'jest';
import renderer, { act } from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Button, IconButton } from '../index';

const RightArrow = () => (
  <svg width="12" height="12" viewBox="0 0 6 10" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M1 9L5 5L1 1" stroke="white" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);

describe('Button', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const root = createRoot(div!);
    act(() => {
      root.render(<Button>Button</Button>);
      root.unmount();
    });
  });

  it('renders button correctly', () => {
    const { getByTestId } = render(
      <Button variant="primary">Primary Button</Button>
    );
    const button = getByTestId('button');

    expect(button).toHaveTextContent('Primary Button');
  });

  it('matches snapshot: primary with icon on left', () => {
    const tree = renderer.create(
      <Button
        variant="primary"
        iconLeft={<RightArrow />}
      >
        Primary Button
      </Button>
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: secondary with icon on right', () => {
    const tree = renderer.create(
      <Button
        variant="secondary"
        iconRight={<RightArrow />}
      >
        Secondary Button
      </Button>
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: text with icon on both sides', () => {
    const tree = renderer.create(
      <Button
        isText
        iconRight={<RightArrow />}
        iconLeft={<RightArrow />}
      >
        Text Button
      </Button>
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: text - disabled and loading', () => {
    const tree = renderer.create(
      <Button
        isText
        disabled
        isLoading
      >
        Text only Button
      </Button>
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: text and custom styles', () => {
    const tree = renderer.create(
      <Button
        isText
        css={{
          padding: '10px'
        }}
      >
        Link text
      </Button>
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('IconButton', () => {
  it('renders icon button correctly', () => {
    const component = renderer.create(<IconButton icon={<RightArrow />} aria-label="Right arrow icon" />);
    expect(component).toMatchSnapshot();
  });
});
