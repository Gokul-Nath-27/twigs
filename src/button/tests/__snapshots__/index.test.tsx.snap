// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot: primary with icon on left 1`] = `
<button
  className="twigs-c-kZaMBa twigs-c-kZaMBa-kyEbhl-size-sm twigs-c-kZaMBa-jgyhUs-variant-primary"
  data-testid="button"
>
  <span
    className="twigs-c-UazGY twigs-c-UazGY-ijxSBfJ-css"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </span>
  Primary Button
</button>
`;

exports[`Button matches snapshot: secondary with icon on right 1`] = `
<button
  className="twigs-c-kZaMBa twigs-c-kZaMBa-kyEbhl-size-sm twigs-c-kZaMBa-fyuIIR-variant-secondary"
  data-testid="button"
>
  Secondary Button
  <span
    className="twigs-c-UazGY twigs-c-UazGY-ieSlpoK-css"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </span>
</button>
`;

exports[`Button matches snapshot: text - disabled and loading 1`] = `
<button
  className="twigs-c-kZaMBa twigs-c-kZaMBa-kyEbhl-size-sm twigs-c-kZaMBa-jgyhUs-variant-primary twigs-c-kZaMBa-hlDObG-isText-true twigs-c-kZaMBa-feQjQZ-cv"
  data-testid="button"
  disabled={true}
>
  <div
    className="twigs-c-PJLV twigs-c-dMQBBi"
    data-testid="dot-loader"
  >
    <div
      className="twigs-c-PJLV"
      data-testid="box"
    />
    <div
      className="twigs-c-PJLV"
      data-testid="box"
    />
    <div
      className="twigs-c-PJLV"
      data-testid="box"
    />
  </div>
</button>
`;

exports[`Button matches snapshot: text and custom styles 1`] = `
<button
  className="twigs-c-kZaMBa twigs-c-kZaMBa-kyEbhl-size-sm twigs-c-kZaMBa-jgyhUs-variant-primary twigs-c-kZaMBa-hlDObG-isText-true twigs-c-kZaMBa-feQjQZ-cv twigs-c-kZaMBa-ihBcVkj-css"
  data-testid="button"
>
  Link text
</button>
`;

exports[`Button matches snapshot: text with icon on both sides 1`] = `
<button
  className="twigs-c-kZaMBa twigs-c-kZaMBa-kyEbhl-size-sm twigs-c-kZaMBa-jgyhUs-variant-primary twigs-c-kZaMBa-hlDObG-isText-true twigs-c-kZaMBa-feQjQZ-cv"
  data-testid="button"
>
  <span
    className="twigs-c-UazGY twigs-c-UazGY-ijxSBfJ-css"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </span>
  Text Button
  <span
    className="twigs-c-UazGY twigs-c-UazGY-ieSlpoK-css"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </span>
</button>
`;
